### DISCUSSION TITLE
actions possible using only toolbox or rest api

### DISCUSSION THREAD

**Discussion by younid** [Keyfactor/ejbca-ce#231] [Q&A] [answered]
Hello,

Can you please tell me if we are able to perform all actions we can do with the admin web site if we use the toolbox cli or the rest api ?
The goal of my question is to know if I will manage to use scripts to create my pki and manage it.
This is for an automated test plateform I need to create.

Is there any tasks that we can't do without the admin web site ?

Thank you in advance for your help.

Regards

**Comment by primetomas** [comment, marked as answer]
You can automate almost everything in EJBCA. I recommend you to follow our Ansible playbooks repo.
https://github.com/Keyfactor/ansible-ejbca-signserver-playbooks

For maximum automation, in EJBCA Enterprise there is ConfigDump, which gives you export/import of all settings in human readable YAML formal.

You can automated the test environment, with things like Ansible and Ansible tower, a click and have a new (production like) test environment up for a dev in 10 minutes, or using containers, or using helm charts.

In the last EJBCA Community meetup in Stockholm there was a presentation and demo on automation, we also have held webinars on this. You can find some parts of it (but not the Ansible webinar I think?) here:
https://www.youtube.com/@KeyfactorCommunity

On the upcoming Tech Days in Barcelona you will also have a chance to see more of it in action (and talk to the devs if you come on-site). https://techdays-2023.keyfactor.com/

Cheers,
Tomas

**Reply by younid** [reply]
Hello @primetomas ,

Thank you for this answer.
Can I ask you if there's a diference between what we can do with the Entreprise and CE edition.
I have only the CE edition for the moment, do I must buy the Entreprise edition if I want to create my test PKI using scripts ?
What are the restriction/limitation of the CE edition on rest api or toolbox ?
I'm asking that but hope you'll just confirm what is said in the playbook link you provided :

> Both Community and Enterprise versions of EJBCA are supported.

Regards

**Reply by primetomas** [reply]
Yes to both.
- Ansible playbooks support both Community and Enterprise
- Enterprise has even richer capabilities for automation with ConfigDump, and additional Management REST APIs, and playbooks can take advantage of these to automate larger multi-component environments such as distributed CA-VA-RA.

Basic differences between Community and Enterprise is here: https://www.primekey.com/community-software/differences-community-vs-enterprise/

The documentation highlights what are specific Enterprise features with the "enteprise" tag, such as:
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/command-line-interfaces/configdump-tool
And in the list of REST API end points here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/protocols/ejbca-rest-interface
